### PR TITLE
Roster management page

### LINKED ISSUE #988: Rosters: Ship roster management UI
Teachers with a roster will want to be able to manage it somehow.

This doesn't need to be "staffshipped" since it will only be accessable for classrooms with rosters, and we'll be staffshipping the creation of rosters.

Planned implementation: (https://github.com/education/classroom/issues/448#issuecomment-311516206)

> Once a roster is uploaded, there will be an option to manage the roster from the classroom settings. The roster management UI is a 2-tabbed list. One tab is "Students" representing all students in a classroom. This includes students that have, or have not linked their GitHub account by accepting an assignment. The other tab is for GitHub accounts that haven't been linked to a student. In this menu, teachers will be able to unlink and link students and GitHub users. This management page will also have the functionality to create or delete students, incase students drop the course or students join late.

### PR DESCRIPTION
**Pull request by nwoodthorpe**
Closes #988, #991 

This is V1 of the roster management page. It allows teachers to:
- View all students in their roster, and whether each one is associated to a GitHub account or not
- View all GitHub accounts who've accepted assignments but have not linked to a roster entry
- Link students to GitHub accounts, and vice versa
- Unlink

I want to ship this as is and introduce the rest of the required features later (this PR is getting big)
- Add roster entries
- Delete roster entries
- Disable unlinked GitHub accounts tab if there are none.
- View / edit roster identifier_name

This is behind a feature flag, non-staff won't see the link to this page, and will 404 if they manually type the URL.

screen shot 2017-07-10 at 9 12 21 am

[Gif of linking from students tab](http://g.recordit.co/LPfbPDqvEt.gif)
[Gif of linking from GitHub accounts tab](http://g.recordit.co/TNNTxtVoTj.gif)
[Gif of unlinking tab](http://g.recordit.co/Abq7OD87Av.gif)

/cc @johndbritton 
/code review @wilfriedE @tarebyte 
/design review @jasonlong 
/design / copy review @mozzadrella

### CODE REVIEW

**Comment by nwoodthorpe** [comment]
^ Ignore all that noise, no violations now that master is merged in.

This is V1 of the roster management page. It allows teachers to:
- View all students in their roster, and whether each one is associated to a GitHub account or not
- View all GitHub accounts who've accepted assignments but have not linked to a roster entry
- Link students to GitHub accounts, and vice versa
- Unlink

I want to ship this as is and introduce the rest of the required features later (this PR is getting big)
- Add roster entries
- Delete roster entries

This is behind a feature flag, non-staff won't see the link to this page, and will 404 if they manually type the URL.

screen shot 2017-07-10 at 9 12 21 am

[Gif of linking from students tab](http://g.recordit.co/LPfbPDqvEt.gif)
[Gif of linking from GitHub accounts tab](http://g.recordit.co/TNNTxtVoTj.gif)
[Gif of unlinking tab](http://g.recordit.co/Abq7OD87Av.gif)

/cc @johndbritton 
/code review @wilfriedE @tarebyte 
/design review @jasonlong 
/design / copy review @mozzadrella

**Comment by jasonlong** [comment]
Thanks @nwoodthorpe. Here's some initial feedback:

The tab labels ("Students" and "Unlinked GitHub accounts") read a little awkwardly to me – e.g. it kind of sounds like unlinked accounts aren't students. Maybe just saying "All students" for the first tab would make it clearer? Def. open to input from others here.

The link and unlink buttons look so similar, it's hard to scan that view and tell the state of each student.

screen shot 2017-07-10 at 1 14 53 pm

Maybe we could try `btn-danger` on the unlink buttons. If those look too heavy next to the outline buttons, we could use regular gray buttons on this view.

And then on the linking action, I don't think we really have any select patterns like this:

screen shot 2017-07-10 at 1 15 46 pm

I wonder if we could try a select menu here. This would have the added benefit of still knowing which row/user you're on as you make your selection.

Something like this (w/o the avatars and label of "Link to student"):

screen shot 2017-07-10 at 1 34 08 pm

**Comment by nwoodthorpe** [comment]
Thanks for the review :)

"All Students" sounds good for the first label. Technically, the accounts in the "Unlinked GitHub accounts" may not be students. These are GitHub accounts who've accepted assignments but haven't linked themselves to a real-life identity yet.

I'll give `btn-danger` a go. I agree though, they do look very similar.

Also yeah, I sort of invented a flow for the linking modal, I wasn't sure which component really fit. The select menu looks good though, I'll give it a shot.

I'll give you another ping when it's all implemented. Cheers!

**Comment by nwoodthorpe** [comment]
@jasonlong I looked into using the GitHub select menu but it's powered by a JS module from `github/github` and CSS that I don't think is in Primer. Would you suggest using the Primer select instead? (http://primercss.io/forms/#selects) or I could try to replicate the GitHub select menu.

**Comment by jasonlong** [comment]
Ah yeah, good point. I don't think a regular select menu makes the most sense here, so let's stick with your current direction. 

Maybe instead of the email address being inside outline buttons, they could just be regular links. If that list were to get fairly long, all those boxes would get distracting. With a little padding between each address, I think we'd find a nice balance of readability and usability.

**Comment by nwoodthorpe** [comment]
@jasonlong Hows this?

screen shot 2017-07-10 at 3 27 32 pm

**Comment by jasonlong** [comment]
That's looking better @nwoodthorpe. Could we kill the underline and the indentation too?

**Comment by nwoodthorpe** [comment]
screen shot 2017-07-11 at 4 36 06 pm

Hows this? @jasonlong

**Comment by jasonlong** [comment]
LGTM!

**Comment by nwoodthorpe** [comment]
This is good for review now

File: app/view_models/unlinked_user/show_view.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module UnlinkedUser
```
Comment (tarebyte) [inline app/view_models/unlinked_user/show_view.rb]: @nwoodthorpe could we add sanity tests for all of these new view models?

Comment (nwoodthorpe) [inline app/view_models/unlinked_user/show_view.rb]: Whoa right... we can spec our view models... *mindblow*

**Comment by nwoodthorpe** [comment]
:raised_hands:, will deploy on Monday

**Comment by nwoodthorpe** [comment]
Tested on prod, looks good :+1:

Also Hound won't run, so linted manually.

```
➜  classroom git:(roster_management_page) script/linter
==> Running rubocop...
Inspecting 246 files
..............................................................................................................................................................................................
........................................................

246 files inspected, no offenses detected
==> Running scss-lint...
```

**Pull request merged**